File: src/main/java/xyz/nkomarn/composter/network/protocol/packet/clientbound/ClientboundLoginPacket.java
package xyz.nkomarn.composter.network.protocol.packet.clientbound;

import xyz.nkomarn.composter.network.WrappedBuff;

public class ClientboundLoginPacket implements ClientboundPacket {

    private final int entityId;
    private final long seed;
    private final byte dimension;

    public ClientboundLoginPacket(int entityId, long seed, byte dimension) {
        this.entityId = entityId;
        this.seed = seed;
        this.dimension = dimension;
    }

    @Override
    public void encode(WrappedBuff buff) {
        buff.writeInt(entityId)
                .writeString("")
                .writeLong(seed)
                .writeByte(dimension);
    }
}

File: src/main/kotlin/kyta/composter/protocol/packet/login/ClientboundLoginPacket.kt
package kyta.composter.protocol.packet.login

import kyta.composter.protocol.Packet
import kyta.composter.protocol.PacketSerializer
import kyta.composter.protocol.WriteBuffer
import kyta.composter.world.dimension.DimensionType

data class ClientboundLoginPacket(
    val entityId: Int,
    val serverId: String,
    val seed: Long,
    val dimension: DimensionType,
) : Packet {
    companion object : PacketSerializer<ClientboundLoginPacket> {
        override fun serialize(packet: ClientboundLoginPacket, buffer: WriteBuffer) {
            buffer.writeInt(packet.entityId)
            buffer.writeString(packet.serverId)
            buffer.writeLong(packet.seed)
            buffer.writeByte(packet.dimension.id)
        }
    }
}